desktop/cmp/rest/impl/RestFormModel.js

```diff
@@ -83,11 +84,13 @@ export class RestFormModel {
         }
 
         if (warning) {
-            const confirmed = await XH.confirm({
-                message: warning,
-                title: 'Warning',
-                icon: Icon.warning({size: 'lg'})
-            });
+            const message = isFunction(warning) ? warning(this.currentRecord) : warning,
```

Comment:
see note above about just always passing in an array of records, even if it is just a single record (or zero records) for some actions